### PR TITLE
fix(drive): panic if PlatformState has serialisation error

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

`PlatfromState.fingerprint` panics in case of serialization error.

## What was done?

- Returned error instead of `expect`

## How Has This Been Tested?

Running tests

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by ogabrielides — APPROVED** [review]

File: packages/rs-drive-abci/src/platform_types/platform_state/mod.rs [hunk 1]
```diff
@@ -103,11 +103,8 @@ impl PlatformDeserializableFromVersionedStructure for PlatformState {
 
 impl PlatformState {
     /// Get the state fingerprint
-    pub fn fingerprint(&self) -> [u8; 32] {
-        hash_double(
-            self.serialize_to_bytes()
-                .expect("expected to serialize state"),
```
[the remark this answers is not in the document]
Comment (ogabrielides) [reply packages/rs-drive-abci/src/platform_types/platform_state/mod.rs:109]: We ran into this today on a devnet due to a bug I introducing while testing in PR #1937

[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/platform_types/platform_state/mod.rs:109]: While I don't really mind this PR, can you explain what was introduced?

[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-drive-abci/src/platform_types/platform_state/mod.rs:109]: We defined PlatformStateForSaving v2 and didn't handle it properly in one of try_from_platform_versioned so we got VersionMismatch error that fails serialisation.

**Comment by QuantumExplorer** [comment]
merging this one in without CI check, as it is very simple.

**Pull request merged**